tools/ddprobe.cpp

```diff
@@ -107,9 +229,24 @@ test_dxgi_duplication(dxgi::adapter_t &adapter, dxgi::output_t &output) {
   // Ensure we can duplicate the current display
   syncThreadDesktop();
 
-  // Return the result of DuplicateOutput() to Sunshine
+  // Attempt to duplicate the output
   dxgi::dup_t dup;
-  return output1->DuplicateOutput((IUnknown *) device.get(), &dup);
+  ComPtr<ID3D11Device> device_ptr(device.get());
+  HRESULT result = output1->DuplicateOutput(device_ptr.Get(), &dup);
+
+  if (FAILED(result)) {
+    std::cout << "Failed to duplicate output [0x"sv << util::hex(result).to_string_view() << "]" << std::endl;
+    return result;
+  }
+
+  // If duplication is successful, test frame capture
+  HRESULT captureResult = test_frame_capture(dup, device_ptr.Get());
+  if (FAILED(captureResult)) {
```

Comment:
`S_FALSE` is a success value, so I'm not sure how this could be working. I think we would just return `S_OK` in the case like we did before?

Reply:
Nice catch! Did not see that at all, fixed it